### DISCUSSION TITLE
v0.10.0-beta2

### DISCUSSION THREAD

**Discussion by laipz8200** [langgenius/dify#8967] [Releases]
## ✨ What’s New in v0.10.0-beta2? ✨

> [!IMPORTANT] 
> This version is for preview only, do not use in production!

> [!NOTE]
> We highly value your feedback! Please feel free to share your thoughts and discuss any issues using the GitHub Discussions feature. You can find the entry point for Discussions at the bottom of the release page.

### 🛠️ Bug Fixes

- **Docker Startup Issue**: Resolved an issue where Dify 0.10.0-beta1 could not be started using Docker.

### Quick Start

To upgrade to this version, follow these steps:

1. Clone the repository using the following command:

   ```bash
   git clone https://github.com/langgenius/dify.git
   ```

2. Switch to the beta branch:

   ```bash
   git checkout 0.10.0-beta2
   ```

3. Then follow the [Deploy with Docker Compose > Starting Dify](https://docs.dify.ai/getting-started/install-self-hosted/docker-compose#starting-dify).

We hope you find the new file processing capabilities and nodes useful in this release. Your feedback and contributions are always appreciated to help us improve Dify!

<hr /><em>This discussion was created from the release <a href='https://github.com/langgenius/dify/releases/tag/0.10.0-beta2'>v0.10.0-beta2</a>.</em>

**Comment by semtex1987** [comment]
Ran a git checkout of a clone copy of my normal dify deployment, to switch it to this 0.10.0-beta2 version, but bringing up docker compose with docker compose up -d there appears to be issues with postgres and a variety of other errors that prevent the stack from successfully deploying

`[SQL: SELECT apps.id, apps.tenant_id, apps.name, apps.description, apps.mode, apps.icon_type, apps.icon, apps.icon_background, apps.app_model_config_id, apps.workflow_id, apps.status, apps.enable_site, apps.enable_api, apps.api_rpm, apps.api_rph, apps.is_demo, apps.is_public, apps.is_universal, apps.tracing, apps.max_active_requests, apps.created_by, apps.created_at, apps.updated_by, apps.updated_at, apps.use_icon_as_answer_icon 
FROM apps 
WHERE apps.tenant_id = %(tenant_id_1)s::UUID AND apps.is_universal = false ORDER BY apps.created_at DESC 
 LIMIT %(param_1)s OFFSET %(param_2)s]
[parameters: {'tenant_id_1': 'xxxxx', 'param_1': 30, 'param_2': 0}]
(Background on this error at: https://sqlalche.me/e/20/f405)
2024-10-01 14:21:37,707.707 ERROR [Dummy-1] [app.py:838] - Exception on /console/api/apps [GET]
Traceback (most recent call last):
  File "/app/api/.venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
  File "/app/api/.venv/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 941, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UndefinedTable: relation "apps" does not exist
LINE 2: FROM apps 
             ^
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
`

**Reply by crazywoola** [reply]
Check this variable `MIGRATION_ENABLED`.

**Reply by semtex1987** [reply]
When enabled, migrations will be executed prior to application startup
 and the application will start after the migrations have completed.
MIGRATION_ENABLED=true

**Comment by weijinqiu** [comment]
1.beta2工作流启动后上传文件，开始节点输出的是一个object的对象，但是值在后面传输的时候使用的却是file类型
![Snipaste_2024-10-09_08-31-58](https://github.com/user-attachments/assets/145f4c75-c228-4021-9a8e-70efd9ffb91f)

2.在file变量作为后面节点的输入变量时，直接传送file变量的name，size，url等内容的时候，后面节点的输入变量会为None

**Reply by laipz8200** [reply]
Thank you for the report!

1. The `File` is displayed in the log in the form of an object, but it cannot be used like an object.
2. We cannot always guarantee obtaining the File’s name, size (especially when the file comes from external sources), and URL (when the file is uploaded locally). However, we will optimize this later to at least ensure that the returned types match what is shown on the interface.

**Reply by weijinqiu** [reply]
这样是file类型的话，当我按照openApi接口规范进行导入自定义工具的时候，需要传入文件，目前生成的自定义工具好像不支持文件类型的选项

**Reply by laipz8200** [reply]
We will actively consider supporting custom tools based on OpenAPI. However, since this feature requires transferring files over the network, it might not be available very soon.
